lib/srv/exec.go

```diff
@@ -165,20 +165,27 @@ func prepareCommand(ctx *ctx) (*exec.Cmd, error) {
 	if ctx.isTestStub {
 		shell = "/bin/sh"
 	}
-	// try to determine the host name of the 1st available proxy to set a nicer
-	// session URL. fall back to <proxyhost> placeholder
-	proxyHost := "<proxyhost>"
+
+	// try to get the address the proxy server can be accessed at. if public_address
+	// is set in the configuration use that, if not try to fall back to the 1st
+	// available proxy from the auth server heartbeat.
+	proxyHost := "<proxyhost>:3080"
 	if ctx.srv != nil {
-		proxies, err := ctx.srv.authService.GetProxies()
+		proxyHost, err = ctx.srv.ProxyPublicAddr()
```

Comment:
this will never be true for server, because exec is only available on proxies. This will always return error.